### ISSUE TITLE
UUID's lingering

### ISSUE THREAD

**Issue by acdoll** [ArctosDB/arctos#2240]
The UUID's that are created when adding additional attribute data through Data Entry remain attached to the specimen after the attributes have been validated and loaded (e.g. http://arctos.database.museum/guid/DMNS:Mamm:19771). I thought these were automatically removed after those attributes are loaded; do we have to do it manually?

**Comment by dustymc** [comment]
There's no way I can know if there's extra data in some Excel file on your desktop linked by UUID, and therefore there's never a safe time when I could remove them.

You can use the OtherID unloader if you want them gone, or just leave them alone - they might be useful for something at some point (like when you find that Excel file full of UUIDs on your desktop!).

**Comment by acdoll** [comment]
Okay; I thought that might be the case. For us, once those attributes are loaded we don't really have any need for that number (at least currently). I am a little concerned about leaving a meaningless number in there for future generations. I'm imagining them wasting time trying to track down it's significance when it doesn't appear anywhere here in the physical world. 
We can certainly use the unloader on a periodic basis, but I still find myself wishing there was an easier way. Could there be a button to 'remove UUIDs' on the Bulkload Attribute page after loading the attributes? This could then run the unloader based on the info just loaded.

**Comment by dustymc** [comment]
Yes, there can certainly be tools. The implications of using them should be clearly documented somewhere.

**Comment by campmlc** [comment]
Reopening. I strongly support making these uuids go away once the associated extras are loaded. I have filed a previous issue documenting the problem that collections personnel have in that we have no way of knowing what "extra" a particular uuid is attached to, in order to confirm or deny whether the associated data have been loaded.  We don't have an Excel file somewhere - these ids are autogenerated by Arctos, and as far as I know we don't have access to them. If they exist in the extras bulkloader, we don't know what collection object they are supposed to link to, because that info is not available in those forms.  IF we could add an other ID to the extras bulkloader, and then download csv from there, we could have a file that documents what uuids are supposed to go with what records. Possible?

**Comment by campmlc** [comment]
Yes, but the data in the loaders has no information that can tie a "extra"
to a collection object that has not yet been cataloged. If we had a field
of associated data, e.g. and OtherID that was entered into the same record
during data entry, then we could download the csv and keep track of which
eventual catalog records had their extras added, and which did not.

**Comment by dustymc** [comment]
The UUID exists entirely to 'tie a "extra" to a collection object that has not yet been cataloged.'

You can download UUID with everything else, including any additional OtherIDs, from the specimen bulkloader.

I think I need an example or something, I don't understand what you're getting at.

**Comment by campmlc** [comment]
Here is a csv download fof extras attributes created before the component loader system in the single record data entry page. I have no way of knowing what collection object they were supposed to link to, to find and resolve the problem, because the only identifier provided here is the uuid. If there were an OtherID , e.g a collector or NK number, of the "parent" record, I could potentially figure these out.

[BulkloadAttributesData_specimen_not_found.zip](https://github.com/ArctosDB/arctos/files/6393516/BulkloadAttributesData_specimen_not_found.zip)

**Comment by dustymc** [comment]
>before the component loader system

There are some significant changes in there, I'm not sure how useful that can be for guiding current development.

This looks like a case of someone having removed the link. There's a bulk tool now, you can remove great swaths of anything you wish, but it should not be surprising when removing the link to data results in data that can no longer be linked.

**Comment by campmlc** [comment]
I personally did not remove anything - so if the link disappeared, perhaps
this is related to the Arctos crash? How is this problem avoided in the new
system?

**Comment by acdoll** [comment]
@campmlc we have had a few issues similar to the example you provided. I believe what happened is that someone (me??) deleted the UUID manually from the catalog record before we had loaded all of the extra data from the separate bulkloaders. I believe this is exactly why Dusty recommended just keeping the UUID's in there. I wonder if we can have the extra loaders in the single entry data page autofill in the GUID entered in that for like the Copy2all function? I realize the GUID is not really created at that point, and something could go wrong before it is, but at least it would leave us a trace to backtrack any orphaned data.

**Comment by dustymc** [comment]
>How is this problem avoided in the new system?

<img width="562" alt="Screen Shot 2021-04-28 at 10 21 35 AM" src="https://user-images.githubusercontent.com/5720791/116446171-88e8d900-a80b-11eb-9a1e-c0a1bc87bd8b.png">


UUIDs are in a dedicated and stable field.

**Comment by campmlc** [comment]
Yes, but if we could autofill with any OtherID in the parent form, this would allow us to backtrack and solve problems such as these. I don't need the catalog number necessarily, in fact, something linking to field notes would be preferable.

**Comment by dustymc** [comment]
I don't think anything else can be remotely as stable as a UUID generated for this purpose at this point. There are very many reasons why a predicted GUID might not actually end up on the record (someone changes it for some reason because it's a "smart" identifier, someone uses that number first by cloning or something, it gets used by two people doing data entry simultaneously, etc., etc., etc.). Any other otherID has all of those possibilities, plus the virtual certainty of not being unique (and so not useful for this) in the first place. All of the alternatives to UUID just look like a lot more work for a lot worse data to me. What exactly is the current actual problem we're trying to solve by adding complication?

**Comment by campmlc** [comment]
The problem is that anyone looking at the data in the following form has no means of going back and finding the collection object that it was supposed to append to if there is a problem with the cataloged item or the linking of the uuid to it. I don't know how to say this more clearly:
![image](https://user-images.githubusercontent.com/14808196/116448475-67d9b580-a816-11eb-9956-62396d575ae7.png)
